app/Services/StandService.php

```diff
@@ -337,104 +343,200 @@ function (Builder $query) use ($airfield) {
+        $standOccupationsToUpdate = new Collection();
+
+        foreach ($aircraftThatCanOccupyStands as $aircraft) {
+            // Still occupying that same stand, nothing to do.
+            $currentlyOccupiedStand = $aircraft->occupiedStand->first();
+            if (
+                $currentlyOccupiedStand &&
+                $this->standOccupied($aircraft, $currentlyOccupiedStand)
+            ) {
+                continue;
             }
-            return null;
-        }
 
-        /*
-         * The aircraft is still occupying its stand.
-         */
-        if ($aircraft->occupiedStand->first() && $this->standOccupied($aircraft, $aircraft->occupiedStand()->first())) {
-            return $aircraft->occupiedStand->first();
+            $selectedStand = $this->getOccupiedStand($aircraft);
+            if ($this->occupiedStandShouldBeRemoved($currentlyOccupiedStand, $selectedStand)) {
+                $occupiedStandsToRemove->add($aircraft);
+            } elseif ($this->occupiedStandShouldBeUpdated($currentlyOccupiedStand, $selectedStand)) {
+                $standOccupationsToUpdate->add(
+                    tap(
+                        $aircraft,
+                        function (NetworkAircraft $aircraft) use ($selectedStand) {
+                            $aircraft->standToBeOccupied = $selectedStand;
+                        }
+                    )
+                );
+            }
         }
 
-        // If there's a stand that's viable, occupy the stand.
-        if ($selectedStand = $this->getOccupiedStand($aircraft)) {
-            $this->occupyStand($aircraft, $selectedStand);
-        }
+        $this->vacateStands($occupiedStandsToRemove);
+        $this->occupyStands($standOccupationsToUpdate);
+    }
 
-        return $selectedStand;
+    /**
+     * Vacate stands where required and trigger the vacation event.
+     */
+    private function vacateStands(Collection $aircraftToVacate): void
+    {
+        // Delete all the allocations
+        DB::table('aircraft_stand')
+            ->whereIn('callsign', $aircraftToVacate->pluck('callsign'))
+            ->delete();
     }
 
     /**
-     * Get the stand that the aircraft is closest to, within occupation parameters.
+     * Occupy the stands that are newly occupied, and remove conflicting assignments.
      */
-    private function getOccupiedStand(NetworkAircraft $aircraft): ?Stand
+    private function occupyStands(Collection $standsToOccupy): void
     {
-        $selectedStand = null;
-        $selectedStandDistance = PHP_INT_MAX;
+        // Update all the occupations
+        $mappedOccupations = $standsToOccupy->map(function (NetworkAircraft $aircraft) {
+            return [
+                'latitude' => $aircraft->latitude,
+                'longitude' => $aircraft->longitude,
+                'callsign' => $aircraft->callsign,
+                'stand_id' => $aircraft->standToBeOccupied->id,
```

Comment:
Can you eager load the standToBeOccupied?

Reply:
There's a bit of trickery in play here. This is a pseudo-attribute that doesn't really exist on the table, The "selected stand" which has already been loaded from `getAllStandsByAirfield` model gets attached to the aircraft model for anti-faff processing :) So in a sense, is already eager loaded!